pkg/cloud/vsphere/actuators/cluster/actuator.go

```diff
@@ -178,25 +191,40 @@ func (a *Actuator) reconcileReadyState(ctx *context.ClusterContext) error {
 	// List the target cluster's nodes to verify the target cluster is online.
 	client, err := remotev1.NewClusterClient(a.controllerClient, ctx.Cluster)
 	if err != nil {
-		ctx.Logger.V(6).Info("unable to get client for target cluster", "reason", err.Error())
+		ctx.Logger.Error(err, "unable to get client for target cluster")
 		return &clusterErr.RequeueAfterError{RequeueAfter: constants.DefaultRequeue}
 	}
 	coreClient, err := client.CoreV1()
 	if err != nil {
-		ctx.Logger.V(6).Info("unable to get core client for target cluster", "reason", err.Error())
+		ctx.Logger.Error(err, "unable to get core client for target cluster")
 		return &clusterErr.RequeueAfterError{RequeueAfter: constants.DefaultRequeue}
 	}
 	if _, err := coreClient.Nodes().List(metav1.ListOptions{}); err != nil {
-		ctx.Logger.V(6).Info("unable to list nodes for target cluster", "reason", err.Error())
+		ctx.Logger.Error(err, "unable to list nodes for target cluster")
 		return &clusterErr.RequeueAfterError{RequeueAfter: constants.DefaultRequeue}
 	}
 
+	// Delete the control plane config map.
+	if err := ctx.CoreClient.ConfigMaps(ctx.Cluster.Namespace).Delete(actuators.GetNameOfControlPlaneConfigMap(ctx.Cluster.UID), &metav1.DeleteOptions{}); err != nil {
+		if !apierrors.IsGone(err) && !apierrors.IsNotFound(err) {
+			ctx.Logger.Error(err, "error deleting the control plane config map")
+			return &clusterErr.RequeueAfterError{RequeueAfter: constants.DefaultRequeue}
+		}
+	}
+
+	// Get the RESTConfig in order to parse its Host to use as the control plane
+	// endpoint to add to the Cluster's API endpoints.
 	restConfig := client.RESTConfig()
 	if restConfig == nil {
-		ctx.Logger.V(6).Info("unable to get rest config target cluster", "reason", err.Error())
+		ctx.Logger.Info("unable to get rest config target cluster")
```

Comment:
Logger.Error?

Reply:
Would have, but there's no error object. I should probably create one and use `Logger.Error` as suggested.